lib/kmer_hash.hh

```diff
@@ -113,6 +116,117 @@ HashIntoType _hash_murmur(const std::string& kmer,
                           HashIntoType& h, HashIntoType& r);
 HashIntoType _hash_murmur_forward(const std::string& kmer);
 
+
+template <typename T, std::size_t N>
```

Comment:
given that this can't work with anything but `unsigned char` at the moment, should we remove the type template argument?